File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        char_count = {}

        if(len(s) != len(t)):
            return False

        for char in s:
            char_count[char] = char_count.get(char,0) + 1

        for char in t:
            if char not in char_count or char_count[char] == 0:
                return False
            char_count[char] -= 1
        return True 
        
             

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char,int> umap;

        for (int i:s)
        {
            umap[i]++;
        }

        for(int i:t)
        {
            umap[i]--;
        }

        for(auto i:umap)
        {
            if(i.second != 0)
            {
                return false;
            }
        }
        return true;
    }
};